File: src/RemoveDuplicatesFromSortedArray.java
public class RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] array) {
        if (array.length == 0) {
            return 0;
        }

        int position = 1;
        for (int index = 1, previouslyEncountered = array[0]; index < array.length ; index++) {
            if (array[index] != previouslyEncountered) {
                array[position++] = array[index];
                previouslyEncountered = array[index];
            }
        }

        return position;
    }
}

File: python/remove_duplicates_from_sorted_array.py
from typing import List


class Solution:
    def removeDuplicates(self, array: List[int]) -> int:
        if len(array) == 0:
            return 0

        position = 1
        previouslyEncountered = array[0]
        for index in range(1, len(array)):
            if array[index] != previouslyEncountered:
                array[position] = array[index]
                position += 1
                previouslyEncountered = array[index]

        return position
